pkg/storage/storage.go

```diff
@@ -193,8 +193,6 @@ func (s *Storage) LockRelease(name string) error {
 // If release doesn't exist or wasn't previously locked - the unlock will pass
 func (s *Storage) UnlockRelease(name string) {
 	s.Log("unlocking release %s", name)
-	s.releaseLocksLock.Lock()
```

Comment:
This still hangs when doing `helm delete` then `Ctrl-C` then `helm delete` but is not detected by `-race`.